### PR TITLE
Merge SCR example codes into SCR_Checkpoint example

### PR DESCRIPTION
**Pull request by nselliott**
# Summary

- This PR is a refactoring
- It does the following:
  - Moves functionality of SCR Write/Read/Output examples into SCR_Checkpoint example.  All except SCR_Checkpoint were only being built, not run, and had some redundant features.  This adds a simple single sequence of saving, loading, and verifying data to the end of the SCR_Checkpoint example and puts all SCR testing into one executable.

### CODE REVIEW

**Review by rhornung67 — APPROVED** [review]
Looks good. Thanks @nselliott

File: src/axom/sidre/examples/spio/IO_SCR_Checkpoint.cpp [hunk 1]
```diff
@@ -321,6 +400,13 @@ int main(int argc, char* argv[])
 
   delete ds;
 
+  // The "simpleTest" calls below do a separate test of a single simple
+  // checkpoint followed by a load from the paralle file system.
```
Comment (agcapps) [inline src/axom/sidre/examples/spio/IO_SCR_Checkpoint.cpp:404]:
```suggestion
  // checkpoint followed by a load from the parallel file system.
```

**Review by agcapps — APPROVED** [review]
Thank you for the consolidation.

**Review by kennyweiss — APPROVED** [review]
Thanks @nselliott

**Pull request merged**